src/expr/src/scalar/func.rs

```diff
@@ -88,6 +88,10 @@ pub enum UnmaterializableFunc {
     IsRbacEnabled,
     MzEnvironmentId,
     MzNow,
+    // Technically this doesn't have to be an unmaterializable function. It can be implemented
+    // fully in SQL by aggregating the mz_role_members catalog table. Unfortunately, we lack the
+    // proper map aggregation functionality.
+    MzRoleOidMemberships,
```

Comment:
You could file two issues: Add the map aggregation type, and convert this to SQL?